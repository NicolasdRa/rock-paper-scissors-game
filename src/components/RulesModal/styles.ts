import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    main: {
      padding: '2rem',
    },

    top: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: '.5rem',
    },

    title: {
      fontSize: '1.8rem',
      fontWeight: 700,
      color: theme.palette.common.darkText,
      textTransform: 'uppercase',
    },

    closeBtn: {
      fontSize: '1.8rem',
      ccolor: theme.palette.common.darkText,
      opacity: 0.5,
    },

    content: { padding: '1rem 2rem 2rem 2rem' },

    btn: {
      position: 'absolute',
      bottom: '2rem',
      right: '2rem',
      padding: '.3rem 2rem',
      color: theme.palette.common.score,
      borderColor: theme.palette.common.score,
      borderRadius: '.5rem',
      '&:hover': {
        color: theme.palette.common.white,
        borderColor: theme.palette.common.white,
      },
    },
  }),
);
