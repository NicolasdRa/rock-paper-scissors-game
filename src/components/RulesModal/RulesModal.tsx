import React, { useState } from 'react';

import {
  Box,
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  IconButton,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

import rules from '../../icons/image-rules.svg';

import { useStyles } from './styles';

export const RulesModal = () => {
  const classes = useStyles();
  const [open, setOpen] = useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <Box className={classes.main}>
      <Button
        variant='outlined'
        color='primary'
        className={classes.btn}
        onClick={handleClickOpen}>
        Rules
      </Button>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'>
        <Box className={classes.top}>
          <DialogTitle disableTypography className={classes.title}>
            Rules
          </DialogTitle>
          <IconButton onClick={handleClose} color='primary'>
            <CloseIcon className={classes.closeBtn} />
          </IconButton>
        </Box>

        <DialogContent className={classes.content}>
          <img src={rules} alt='rules' />
        </DialogContent>
      </Dialog>
    </Box>
  );
};
