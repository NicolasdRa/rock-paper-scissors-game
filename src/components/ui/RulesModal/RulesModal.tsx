import { useState } from "react";
import useMediaQuery from "@material-ui/core/useMediaQuery";

import {
  Box,
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  IconButton,
  useTheme,
} from "@material-ui/core";
import CloseIcon from "@material-ui/icons/Close";

import rules from "../../../icons/image-rules.svg";

import { useStyles } from "./styles";

export const RulesModal: React.FC = () => {
  const classes = useStyles();
  const theme = useTheme();

  const matches = useMediaQuery(theme.breakpoints.down("md"));
  const [open, setOpen] = useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <Box className={classes.main}>
      <Button
        variant="outlined"
        color="primary"
        className={classes.btn}
        onClick={handleClickOpen}
      >
        Rules
      </Button>
      <Dialog
        fullScreen={matches}
        open={open}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
        className={classes.modal}
      >
        <Box className={classes.top}>
          <DialogTitle disableTypography className={classes.title}>
            Rules
          </DialogTitle>
          {!matches && (
            <IconButton onClick={handleClose} color="primary">
              <CloseIcon className={classes.closeBtn} />
            </IconButton>
          )}
        </Box>

        <DialogContent className={classes.content}>
          <img src={rules} alt="rules" />
        </DialogContent>

        {matches && (
          <IconButton onClick={handleClose} color="primary">
            <CloseIcon className={classes.closeBtnPhone} />
          </IconButton>
        )}
      </Dialog>
    </Box>
  );
};
