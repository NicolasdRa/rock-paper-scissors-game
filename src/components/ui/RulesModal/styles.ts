import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    main: {
      padding: "2rem",
    },

    modal: {
      [theme.breakpoints.down("md")]: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-around",
        alignItems: "center",
      },
    },

    top: {
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-between",
      alignItems: "center",
      padding: ".5rem",

      [theme.breakpoints.down("md")]: {
        justifyContent: "center",
        marginTop: "4rem",
      },
    },

    title: {
      fontSize: "1.8rem",

      fontWeight: 700,
      color: theme.palette.common.darkText,
      textTransform: "uppercase",

      [theme.breakpoints.down("md")]: {
        fontSize: "2.5rem",
      },
    },

    closeBtn: {
      fontSize: "1.9rem",
      marginRight: "1rem",
      color: theme.palette.common.darkText,
      opacity: 0.5,
    },

    closeBtnPhone: {
      fontSize: "1.9rem",
      color: theme.palette.common.darkText,
      opacity: 0.5,
      marginBottom: "4rem",
    },

    content: {
      padding: "1rem 2rem 2rem 2rem",

      [theme.breakpoints.down("md")]: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-around",
        alignItems: "center",
      },
    },

    btn: {
      padding: ".3rem 2rem",
      color: theme.palette.common.score,
      borderColor: theme.palette.common.score,
      borderRadius: ".5rem",
      "&:hover": {
        color: theme.palette.common.white,
        borderColor: theme.palette.common.white,
      },
    },
  })
);
