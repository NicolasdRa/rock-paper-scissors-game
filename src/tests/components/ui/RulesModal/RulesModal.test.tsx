import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { RulesModal } from "../../../../components/ui/RulesModal/RulesModal";

describe("tests on <RulesModal/>", () => {
  const wrapper = mount(<RulesModal />);

  test("should render <RulesModal/> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
