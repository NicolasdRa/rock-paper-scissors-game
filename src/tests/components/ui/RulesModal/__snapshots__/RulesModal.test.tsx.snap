// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <RulesModal/> should render <RulesModal/> correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-9 makeStyles-main-1"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-btn-8 MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Rules
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
